src/test/java/kitchenpos/application/MenuServiceTest.kt

```diff
@@ -0,0 +1,518 @@
+                this.price = BigDecimal.TWO
+            })
+
+            // when & then
+            shouldThrowExactly<IllegalArgumentException> {
+                menuService.create(request)
+            }
+        }
+
+        @DisplayName("이름에 욕설이 포함되어 있다면, IllegalArgumentException 예외 처리를 한다.")
+        @Test
+        fun test11() {
+            // given
+            val menuProduct = MenuProduct().apply {
+                this.productId = UUID.randomUUID()
+                this.quantity = 1
+            }
+
+            val request = Menu().apply {
+                this.price = BigDecimal.ONE
+                this.menuGroupId = UUID.randomUUID()
+                this.menuProducts = listOf(menuProduct)
+                this.name = "욕설"
+            }
+
+            every { menuGroupRepository.findById(any()) } returns Optional.of(MenuGroup())
+            every { productRepository.findAllByIdIn(any()) } returns listOf(Product())
+            every { productRepository.findById(any()) } returns Optional.of(Product().apply {
+                this.price = BigDecimal.TWO
+            })
+            every { purgomalumClient.containsProfanity(any()) } returns true
+
+            // when & then
+            shouldThrowExactly<IllegalArgumentException> {
+                menuService.create(request)
+            }
+        }
+
+        @DisplayName("정상 조건이라면, 메뉴를 저장한다.")
+        @Test
+        fun test12() {
+            // given
+            val menuProduct = MenuProduct().apply {
+                this.productId = UUID.randomUUID()
+                this.quantity = 1
+            }
+
+            val request = Menu().apply {
+                this.price = BigDecimal.ONE
+                this.menuGroupId = UUID.randomUUID()
+                this.menuProducts = listOf(menuProduct)
+                this.name = "정상 메뉴"
+                this.isDisplayed = true
+                this.id = UUID.randomUUID()
+                this.menuGroup = MenuGroup()
+            }
+
+            every { menuGroupRepository.findById(any()) } returns Optional.of(MenuGroup())
+            every { productRepository.findAllByIdIn(any()) } returns listOf(Product())
+            every { productRepository.findById(any()) } returns Optional.of(Product().apply {
+                this.price = BigDecimal.TWO
+            })
+
+            every { purgomalumClient.containsProfanity(any()) } returns false
+            every { menuRepository.save(any()) } returns request
+
+            // when
+            val result = menuService.create(request)
+
+            // then
+            verify(exactly = 1) { menuRepository.save(any()) }
+            result.id shouldBe request.id
+            result.name shouldBe request.name
+            result.price shouldBe request.price
+            result.menuGroup shouldBe request.menuGroup
+            result.isDisplayed shouldBe request.isDisplayed
+            result.menuProducts shouldBe request.menuProducts
```

Comment:
사소한 팁을 드리자면,
여러 개의 검증을 해야 할때 assertSoftly { }로 묶어서 한번에 수행할 수도 있습니다.
일반적으로 여러 개의 검증은 진행하다 실패하면 바로 멈추고 후속 검증은 수행하지 않게 되는데
assertSoftly는 모두 검증해서 각 실패한 위치를 모두 알려주는 차이가 있습니다.
```suggestion
            assertSoftly {
                result.id shouldBe request.id
                result.name shouldBe request.name
                result.price shouldBe request.price
                result.menuGroup shouldBe request.menuGroup
                result.isDisplayed shouldBe request.isDisplayed
                result.menuProducts shouldBe request.menuProducts
            }
```

Reply:
해당 방식으로 모두 적용하였습니다!